src/app/page-with-context.tsx

```
@@ -0,0 +1,242 @@
+import Link from 'next/link';
+import { kv } from '@/lib/redis-client';
+import { SiteConfig, Category } from '@/types';
+import { getSiteByHostname, generateSiteBaseUrl, generateCategoryHref } from '@/lib/site-utils';
+import { CategoryLink } from '@/components/LinkUtilities';
+import MainLayout from '@/components/MainLayout';
+
+// Types for Next.js App Router page props
+type PageProps = {
+  params: {};
```

Biome (1.9.4)

[error] Don't use '{}' as a type.

Prefer explicitly define the object shape. '{}' means "any non-nullable value".

(lint/complexity/noBannedTypes)